site/src/pages/GroupsPage/GroupPage.tsx

```diff
@@ -219,14 +188,29 @@ export const GroupPage: React.FC = () => {
                                   menuItems={[
                                     {
                                       label: "Remove",
-                                      onClick: () => {
-                                        send({
-                                          type: "REMOVE_MEMBER",
-                                          userId: member.id,
-                                        });
+                                      onClick: async () => {
+                                        try {
+                                          await removeMemberMutation.mutateAsync(
```

Comment:
I'm not 100% sure about the setup right now, but since there's one mutation in place at the top of the component, would this have the risk of intermingling mutation states? As in, the states for the queries themselves might be okay, but the every mapped group item would be be able to overwrite the mutation as shared mutable state?

I'm not sure if a `mutation.reset` call could help, or if my understanding of RQ might be off here, and this actually is safe

Reply:
What use case you see this causing an issue?

Reply:
I'm mainly wondering about the `removeMemberMutation.mutateAsync` call while it's being awaited, since there'll be a period between the function being called and it resolving where the user can still interact with the UI.

Would we want to let the user keep clicking menu items while a mutation is happening? With one shared mutation, wouldn't we only be able to track the last mutation started, even if other mutations are going on?

Reply:
Thanks for sharing this, it was super helpful 👍